### PR TITLE
fix: wallet connection and chainStorageWatcher

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
- there was a regression in #57 that broke `makeAgoricChainStorageWatcher` - it needs to take an apiUrl instead of rpc
 - `makeAgoricWalletConnection` needs an rpc to work, and an api endpoint, so this change brings it up to date with the new @agoric/rpc changes
 - also ensures there's only one `yarn.lock` in the workspace
 
 
 Tested manually as it'd be a bit of work to add testing with keplr. A more cost effective test in the interim could be a UI banner that shows if the instance was found (API only), and a test to see if it's present. That would validate `chainStorageWatcher` is working (and, the contract install). Edit: Seem something to this effect is tracked here: https://github.com/Agoric/dapp-offer-up/issues/46

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -29,10 +29,12 @@ const terms = {
...
-const watcher = makeAgoricChainStorageWatcher(
-  'http://localhost:26657',
-  'agoriclocal',
-);
+const ENDPOINTS = {
+  RPC: 'http://localhost:26657',
+  API: 'http://localhost:1317',
+};
+
+const watcher = makeAgoricChainStorageWatcher(ENDPOINTS.API, 'agoriclocal');
```
Comment (dckc) [inline ui/src/App.tsx:37]: here's hoping for

 - https://github.com/Agoric/ui-kit/issues/79

**Comment by 0xpatrickdev** [comment]
P.S. if you are testing manually, `voteLatestProposalAndWait` seems to consistently fail - out of gas. 

Revisiting efforts to fix that upstream: https://github.com/Agoric/agoric-3-proposals/pull/73

**Comment by dckc** [comment]
...
> * also ensures there's only one `yarn.lock` in the workspace

any particular reason? I thought separating them was on purpose.

**Review by dckc — APPROVED** [review]
This looks like it can't hurt.

**Comment by 0xpatrickdev** [comment]
> ...
> 
> > * also ensures there's only one `yarn.lock` in the workspace
> 
> any particular reason? I thought separating them was on purpose.

It's my interpretation after reading [this article](https://classic.yarnpkg.com/lang/en/docs/workspaces/#toc-why-would-you-want-to-do-this). When I was updating the `@agoric/web-components` (from the workspace root) I kept observing it having no effect on the ui lock file.

**Comment by dckc** [comment]
I don't think we want to use code from the contract package in the UI nor vice versa. And I'm pretty sure we want to be able to use new shiny stuff in the ui before issues that would complicate compatibility between the contract and the chain get addressed. But I'm OK dealing with all that outside this PR.

**Comment by 0xpatrickdev** [comment]
> I don't think we want to use code from the contract package in the UI nor vice versa. And I'm pretty sure we want to be able to use new shiny stuff in the ui before issues that would complicate compatibility between the contract and the chain get addressed.

In yarn classic I believe there's a [nohoist](https://classic.yarnpkg.com/blog/2018/02/15/nohoist/) key that can be parameterized in `workspaces` in `package.json` to achieve this. In more recent versions of yarn, there's a [nmHoistingLimits](https://yarnpkg.com/configuration/yarnrc#nmHoistingLimits) param that goes in `.yarnrc.yml`.

**Pull request merged**